File: AEDS/aeds2/tp1/palindromo/recursivo/PalindromoRecursivo.java
import java.util.Scanner;

public class PalindromoRecursivo {
    
    //funcao para comparar strings e determinar se sao identicas
    public static boolean isEquals(String palavra1, String palavra2){
        int n = palavra1.length();
        boolean resp = true;

        if(n != palavra2.length()){
            resp = false;
        } else {
            for(int i = 0; i < n; i++){
                if(palavra1.charAt(i) != palavra2.charAt(i)){
                    resp = false;
                    i = n;
                 }
            }
        }
        return (resp);
    }
    public static void classificarPalindromo(String s){
		if (classificarPalindromo(s, 0)) {
            System.out.println("SIM");
        } else {
            System.out.println("NAO");
        }
    }

    // metodo para classificar se uma palavra e' ou nao um palindromo
    public static boolean classificarPalindromo(String s, int i){
    	int n = s.length();
        boolean resp;
		
		// caso base
        if (i >= n / 2) {
            resp = true;
        } else if (s.charAt(i) != s.charAt(n - i - 1)) {
            resp = false;
        } else {
        	// chamada recursiva 
        	resp = classificarPalindromo(s, i + 1);
		}
   
		return (resp);
	}

    // metodo principal
    public static void main (String[] args){
    	String s;
        Scanner sc = new Scanner(System.in);
            
        s = sc.nextLine();
           
        while(!isEquals(s, "FIM")){
        	classificarPalindromo(s);
            s = sc.nextLine();
        }
 		sc.close();         
    }
}

File: 2° semestre/aeds2/tp1/palindromo/c/recursivo/palindromoRecursivo.c
#include <stdio.h>
#include <stdbool.h>

// funcao para determinar tamanho da palavra
int str_len(char *palavra){
    int length = 0;
	while(*(palavra+length)){ 
		length++;
	}
	return(length);
}

//funcao para comparar strings 
bool str_cmp(char palavra1[], char palavra2[]){
    bool resp = true;
    int length = str_len(palavra1);

    if(length != str_len(palavra2)){
        resp = false;    
    } else {
        for(int i = 0; i < length; i++){
            if(palavra1[i] != palavra2[i]){
               resp = false;
               i = length;
            } 
        }
    }
    return (resp);
}

//funcao para determinar se uma palavra e' ou nao palindromo
bool classificarPalindromo(char palavra[], int i){
    int n = str_len(palavra);
    bool resp;
	
	if(i >= n / 2){
		resp = true;
	} else if (palavra[i] != palavra[n - i - 1]){
        resp = false;
    } else {
		resp = classificarPalindromo(palavra, i + 1);
    }

    return resp;
}

//metodo principal
int main (){
    char palavra[500];
    scanf(" %[^\n]", palavra);  
    
    while(!str_cmp(palavra, "FIM")){
        if(classificarPalindromo(palavra, 0))
			printf("SIM\n");
		else
            printf("NAO\n");
        scanf(" %[^\n]", palavra);  
    }
    return 0;
}
